### ISSUE TITLE
gentypesystem: std_vector_x_to_cpp causes segfault if argument not a numpy array

### ISSUE THREAD

**Issue by hodger** [cyclus/cyclus#1347]
I found this issue while working on the Hdf5_back_gen tests, caused a segfault. 

If you add a `VECTOR_INT` as a regular python list to a recorder, like this:
```
datum.add_val('col1', [1,2,3,4,5], shape=[5, -1], dbtype=cpp_typesystem.VECTOR_INT)
```
The Cython referenced is this:
```
cdef std_vector[int] std_vector_int_to_cpp(object x):
    cdef int i
    cdef int x_size
    cdef std_vector[int] cppx
    cdef int * x_data
    x_size = len(x)
    if isinstance(x, np.ndarray) and (<np.ndarray> x).descr.type_num == np.NPY_INT32:
        x_data = <int *> np.PyArray_DATA(<np.ndarray> x) 
        # For numpy arrays we resize
        cppx.resize(<size_t> x_size)
        memcpy(<void*> &cppx[0], x_data, sizeof(int) * x_size)
    else:
        for i, x_val in enumerate(x):
            # Here we index before a resize
            cppx[i] = <int> x_val
    return cppx
```
@scopatz is the fix just adding to the template in `gentypesystem.py`? That seemed to work for me, but I'm not sure if anything else needs to be tweaked

**Comment by scopatz** [comment]
Hi @hodger - yeah, we just need to add a resize, that is right. Probably before the loop since we know the size we want to resize to.

**Comment by hodger** [comment]
@scopatz do you think this is something worthy of throwing into #1346, or should I handle it in my upcoming PR?

**Comment by scopatz** [comment]
I think it would be best to just open a small PR with the fix and then I can merge that in quickly.  Or you can put it in your upcoming PR.  Either way. It doesn't make sense for #1346, since those were interface issues I found while writing docs.

**Comment by hodger** [comment]
Understood, thanks

**Comment by hodger** [comment]
addressed by #1349, closing
